File: sdks/dotnet/Models/Secret.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace VaultAgent.SDK.Models
{
    /// <summary>
    /// Represents a secret with its metadata and value
    /// </summary>
    public class Secret
    {
        [JsonPropertyName("id")]
        public string Id { get; set; }

        [JsonPropertyName("name")]
        public string Name { get; set; }

        [JsonPropertyName("value")]
        public string Value { get; set; }

        [JsonPropertyName("metadata")]
        public Dictionary<string, object> Metadata { get; set; } = new();

        [JsonPropertyName("tags")]
        public List<string> Tags { get; set; } = new();

        [JsonPropertyName("created_at")]
        public DateTime CreatedAt { get; set; }

        [JsonPropertyName("updated_at")]
        public DateTime UpdatedAt { get; set; }

        [JsonPropertyName("expires_at")]
        public DateTime? ExpiresAt { get; set; }

        [JsonPropertyName("rotation_due")]
        public DateTime? RotationDue { get; set; }

        [JsonPropertyName("version")]
        public int Version { get; set; }

        [JsonPropertyName("created_by")]
        public string CreatedBy { get; set; }

        [JsonPropertyName("access_count")]
        public long AccessCount { get; set; }

        [JsonPropertyName("last_accessed")]
        public DateTime? LastAccessed { get; set; }

        [JsonPropertyName("status")]
        public string Status { get; set; }

        public override string ToString()
        {
            return $"Secret{{Id='{Id}', Name='{Name}', Version={Version}, Status='{Status}', CreatedAt={CreatedAt}}}";
        }
    }

    /// <summary>
    /// Represents secret metadata without the actual value
    /// </summary>
    public class SecretMetadata
    {
        [JsonPropertyName("id")]
        public string Id { get; set; }

        [JsonPropertyName("name")]
        public string Name { get; set; }

        [JsonPropertyName("metadata")]
        public Dictionary<string, object> Metadata { get; set; } = new();

        [JsonPropertyName("tags")]
        public List<string> Tags { get; set; } = new();

        [JsonPropertyName("created_at")]
        public DateTime CreatedAt { get; set; }

        [JsonPropertyName("updated_at")]
        public DateTime UpdatedAt { get; set; }

        [JsonPropertyName("expires_at")]
        public DateTime? ExpiresAt { get; set; }

        [JsonPropertyName("rotation_due")]
        public DateTime? RotationDue { get; set; }

        [JsonPropertyName("version")]
        public int Version { get; set; }

        [JsonPropertyName("created_by")]
        public string CreatedBy { get; set; }

        [JsonPropertyName("access_count")]
        public long AccessCount { get; set; }

        [JsonPropertyName("last_accessed")]
        public DateTime? LastAccessed { get; set; }

        [JsonPropertyName("status")]
        public string Status { get; set; }

        public override string ToString()
        {
            return $"SecretMetadata{{Id='{Id}', Name='{Name}', Version={Version}, Status='{Status}', CreatedAt={CreatedAt}}}";
        }
    }

    /// <summary>
    /// Request object for creating a new secret
    /// </summary>
    public class CreateSecretRequest
    {
        [JsonPropertyName("name")]
        public string Name { get; set; }

        [JsonPropertyName("value")]
        public string Value { get; set; }

        [JsonPropertyName("metadata")]
        public Dictionary<string, object> Metadata { get; set; } = new();

        [JsonPropertyName("tags")]
        public List<string> Tags { get; set; } = new();

        public CreateSecretRequest() { }

        public CreateSecretRequest(string name, string value)
        {
            Name = name;
            Value = value;
        }

        public CreateSecretRequest(string name, string value, Dictionary<string, object> metadata, List<string> tags)
        {
            Name = name;
            Value = value;
            Metadata = metadata ?? new();
            Tags = tags ?? new();
        }
    }

    /// <summary>
    /// Request object for updating an existing secret
    /// </summary>
    public class UpdateSecretRequest
    {
        [JsonPropertyName("value")]
        public string Value { get; set; }

        [JsonPropertyName("metadata")]
        public Dictionary<string, object> Metadata { get; set; }

        [JsonPropertyName("tags")]
        public List<string> Tags { get; set; }

        public UpdateSecretRequest() { }

        public UpdateSecretRequest(string value)
        {
            Value = value;
        }
    }

    /// <summary>
    /// Options for listing secrets
    /// </summary>
    public class ListSecretsOptions
    {
        public List<string> Tags { get; set; }
        public int? Limit { get; set; }
        public int? Offset { get; set; }
    }

    /// <summary>
    /// Response object for listing secrets
    /// </summary>
    public class SecretsResponse
    {
        [JsonPropertyName("secrets")]
        public List<SecretMetadata> Secrets { get; set; } = new();
    }
}

File: sdks/java/src/main/java/com/vaultagent/sdk/model/Secret.java
package com.vaultagent.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.Instant;
import java.util.List;
import java.util.Map;

/**
 * Represents a secret with its metadata and value
 */
public class Secret {
    
    @JsonProperty("id")
    private String id;
    
    @JsonProperty("name")
    private String name;
    
    @JsonProperty("value")
    private String value;
    
    @JsonProperty("metadata")
    private Map<String, Object> metadata;
    
    @JsonProperty("tags")
    private List<String> tags;
    
    @JsonProperty("created_at")
    private Instant createdAt;
    
    @JsonProperty("updated_at")
    private Instant updatedAt;
    
    @JsonProperty("expires_at")
    private Instant expiresAt;
    
    @JsonProperty("rotation_due")
    private Instant rotationDue;
    
    @JsonProperty("version")
    private Integer version;
    
    @JsonProperty("created_by")
    private String createdBy;
    
    @JsonProperty("access_count")
    private Long accessCount;
    
    @JsonProperty("last_accessed")
    private Instant lastAccessed;
    
    @JsonProperty("status")
    private String status;
    
    // Constructors
    public Secret() {}
    
    public Secret(String name, String value) {
        this.name = name;
        this.value = value;
    }
    
    // Getters and setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    
    public String getValue() { return value; }
    public void setValue(String value) { this.value = value; }
    
    public Map<String, Object> getMetadata() { return metadata; }
    public void setMetadata(Map<String, Object> metadata) { this.metadata = metadata; }
    
    public List<String> getTags() { return tags; }
    public void setTags(List<String> tags) { this.tags = tags; }
    
    public Instant getCreatedAt() { return createdAt; }
    public void setCreatedAt(Instant createdAt) { this.createdAt = createdAt; }
    
    public Instant getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Instant updatedAt) { this.updatedAt = updatedAt; }
    
    public Instant getExpiresAt() { return expiresAt; }
    public void setExpiresAt(Instant expiresAt) { this.expiresAt = expiresAt; }
    
    public Instant getRotationDue() { return rotationDue; }
    public void setRotationDue(Instant rotationDue) { this.rotationDue = rotationDue; }
    
    public Integer getVersion() { return version; }
    public void setVersion(Integer version) { this.version = version; }
    
    public String getCreatedBy() { return createdBy; }
    public void setCreatedBy(String createdBy) { this.createdBy = createdBy; }
    
    public Long getAccessCount() { return accessCount; }
    public void setAccessCount(Long accessCount) { this.accessCount = accessCount; }
    
    public Instant getLastAccessed() { return lastAccessed; }
    public void setLastAccessed(Instant lastAccessed) { this.lastAccessed = lastAccessed; }
    
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    
    @Override
    public String toString() {
        return "Secret{" +
                "id='" + id + '\'' +
                ", name='" + name + '\'' +
                ", version=" + version +
                ", status='" + status + '\'' +
                ", createdAt=" + createdAt +
                '}';
    }
}